### PR TITLE
Flip the mask in spectral_interpolate

### PR DESCRIPTION
**Pull request by e-koch**

### CODE REVIEW

**Comment by e-koch** [comment]
If `indiff < 0` when spectrally interpolating, the mask should also be flipped before interpolating.

I added a test for this case.

File: spectral_cube/spectral_cube.py [hunk 1]
```diff
@@ -2519,8 +2519,10 @@ def spectral_interpolate(self, spectral_grid,
             cubedata = self.filled_data[::-1]
```
Comment (e-koch) [inline spectral_cube/spectral_cube.py]: @keflavich - Does the whole cube need to be loaded in like this? Is there an advantage to doing so?

Comment (keflavich) [inline spectral_cube/spectral_cube.py]: Hmm... no, now that you mention it, I think we could change this to
```
cubedata = self.filled_data
specslice = slice(None) if indiff >=0 else slice(None,None,-1)
```
then later, use
```
                newcube[:,iy,ix] = np.interp(spectral_grid.value, inaxis.value,
                                             cubedata[specslice,iy,ix].value)
```

**Comment by keflavich** [comment]
Looks like there is one minor thing to fix or at least understand: Under some conditions, this warning isn't being raised?

https://travis-ci.org/radio-astro-tools/spectral-cube/jobs/209818825#L704

**Comment by e-koch** [comment]
I can reproduce it locally. Changing `warnings.simplefilter('default')` to `"always"` in `test_varyres_moment` and `test_varyres_moment_logic_issue364` seems to fix it. I'm unsure why this only happens for one of the test cases, though.

**Comment by keflavich** [comment]
...is this affecting anything before your PR?

**Comment by e-koch** [comment]
Yeah, it's failing on the master branch too. Looks like it has been for awhile: https://travis-ci.org/radio-astro-tools/spectral-cube/builds

**Comment by keflavich** [comment]
We'll fix the other issue in #370

**Pull request merged**